sound/soc/sof/ipc4-topology.c

```diff
@@ -1398,12 +1399,25 @@ static int snd_sof_get_nhlt_endpoint_data(struct snd_sof_dev *sdev, struct snd_s
 					   dir, dev_type);
 
 	if (!cfg) {
+		if (linktype == SOF_DAI_INTEL_DMIC) {
+			if (params_width(params) != 32) {
+				/* look for a blob based on params if there's no 32-bit blob */
+				bit_depth = params_width(params);
```

Comment:
are we going to have similar workaround for rate, channel?

Reply:
I cant imagine those atm. @singalsu what do you think?

Reply:
Channels count to capture should be fixed, there's no way to blindly know what channels to leave out or even extrapolate. But a  stream can be captured with different word lengths and rates depending on quality need.

What if we have SRC in capture topology, variable rate side for PCM, fixed rate (48 kHz) for DAI?

If I understood correctly this patch is for allowing 16/24/32 capture from PCM with host copier format convert while operating DMIC DAI copier with fixed format defined for DAI copier, e.g. S32_LE. I still haven't tried this patch with the issue that we saw with Ranjani, but I will.

Reply:
I seem to need to remove 2ch formats options from 4ch capture to get the wanted 32 bit DSP capture pipeline with option for 16 bit capture to user space. We'd need to duplicate quite a lot tplg2 stuff for 2ch and 4ch separated.